src/actinia_core/core/common/keycloak_user.py

```diff
@@ -0,0 +1,323 @@
+# -*- coding: utf-8 -*-
+#######
+# actinia-core - an open source REST API for scalable, distributed, high
+# performance processing of geographical data that uses GRASS GIS for
+# computational tasks. For details, see https://actinia.mundialis.de/
+#
+# Copyright (c) 2016-2018 Sören Gebbert and mundialis GmbH & Co. KG
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+#######
+
+"""
+Keycloak user management
+
+TODO: list_all_users, delete, create_user (, exists) to be added later
+      (keycloak admin console is needed!)
+"""
+from os.path import isfile
+from json import load as json_load
+from keycloak import KeycloakOpenID
+from jose.exceptions import ExpiredSignatureError
+
+from actinia_core.core.common.user_base import (
+    ActiniaUserBase,
+)
+from actinia_core.core.logging_interface import log
+from actinia_core.core.common.config import global_config
+
+__author__ = "Anika Weinmann"
+__copyright__ = "Copyright 2022, mundialis GmbH & Co. KG"
+__maintainer__ = "mundialis GmbH & Co. KG"
+
+
+def read_keycloak_config(keycloak_config_path=None):
+    global KEYCLOAK_URL, REALM, CLIENT_ID, CLIENT_SECRET_KEY
+    if keycloak_config_path:
+        global_config.KEYCLOAK_CONFIG_PATH = keycloak_config_path
+    if global_config.KEYCLOAK_CONFIG_PATH:
+        if isfile(global_config.KEYCLOAK_CONFIG_PATH):
+            with open(global_config.KEYCLOAK_CONFIG_PATH) as f:
+                keycloak_cfg = json_load(f)
+                KEYCLOAK_URL = keycloak_cfg["auth-server-url"]
+                REALM = keycloak_cfg["realm"]
+                CLIENT_ID = keycloak_cfg["resource"]
+                CLIENT_SECRET_KEY = keycloak_cfg["credentials"]["secret"]
+        else:
+            raise Exception(
+                "KEYCLOAK_CONFIG_PATH is not a valid keycloak configuration "
+                "for actinia"
+            )
```

Comment:
Not really sure about this suggestion, but somehow I find it confusing that config is stored / read in different places. I know that this is only the config read directly from the json generated by keycloak, but maybe it would be more nice to have this section in the config.py, storing the values directly to `global_config` object instead of working with global variables here!?

Reply:
yes I can do it